### PR TITLE
Improve EncoderDecoderModel docs

### LINKED ISSUE #16135: EncoderDecoderModel docs:
## **First good issue**

There have been quite some issues/questions with how to use the Encoder-Decoder model, e.g.: https://github.com/huggingface/transformers/issues/4483 and https://github.com/huggingface/transformers/issues/15479 . The main reason for this is that the model docs are quite outdated and we could need a nice How-to-guide. 

So I think we have two action items here:

1. Improve https://huggingface.co/docs/transformers/v4.17.0/en/model_doc/encoder-decoder#encoder-decoder-models a.k.a.: https://github.com/huggingface/transformers/blob/master/docs/source/model_doc/encoder-decoder.mdx

We should mention here: 
a) How to create a model ? We should show how to use the `from_encoder_decoder_pretrained(...)` and then how to save the model?
b) How to fine-tune this model? We should mention that this model can then be fine-tuned just like any other encoder-decoder model (Bart, T5, ...)
c) Put a big warning that the config values have to be correctly set and how to set them, e.g. read: https://github.com/huggingface/transformers/issues/15479

This should be an `EncoderDecoderModel` specific text and be very concise and short.

In a second step, we should then write a How-to-guide that includes much more details.

More than happy to help someone tackle this first good issue

### PR DESCRIPTION
**Pull request by Ryukijano**
Fixes #16135

Improve the `EncoderDecoderModel` documentation.

* Add example code to create an `EncoderDecoderModel` using `from_encoder_decoder_pretrained`.
* Add instructions on how to save the model.
* Add instructions on how to fine-tune the model.
* Add a warning about correctly setting configuration values.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/huggingface/transformers/issues/16135?shareId=57362139-2e86-4020-90f9-f15402ecf6e3).

### CODE REVIEW

**Comment by LysandreJik** [comment]
cc @stevhliu

File: docs/source/en/model_doc/encoder-decoder.md [hunk 1]
```diff
@@ -178,3 +178,51 @@ were contributed by [ydshieh](https://github.com/ydshieh).
 
 </jax>
 </frameworkcontent>
+
+## How to create a model
+
+To create an `EncoderDecoderModel`, you can use the `from_encoder_decoder_pretrained` method. This method allows you to initialize the model from a pretrained encoder and decoder checkpoint. Here is an example:
```
Comment (stevhliu) [inline docs/source/en/model_doc/encoder-decoder.md:184]:
```suggestion
Use the [`~EncoderDecoderModel.from_encoder_decoder_pretrained`] method to initialize an [`EncoderDecoderModel`] from a pretrained encoder and decoder checkpoint. Here is an example:
```

File: docs/source/en/model_doc/encoder-decoder.md [hunk 2]
```diff
@@ -178,3 +178,51 @@ were contributed by [ydshieh](https://github.com/ydshieh).
 
 </jax>
 </frameworkcontent>
+
+## How to create a model
```
Comment (stevhliu) [inline docs/source/en/model_doc/encoder-decoder.md:182]:
```suggestion
## Create a model
```

File: docs/source/en/model_doc/encoder-decoder.md [hunk 3]
```diff
@@ -178,3 +178,51 @@ were contributed by [ydshieh](https://github.com/ydshieh).
...
+model = EncoderDecoderModel.from_encoder_decoder_pretrained("bert-base-uncased", "bert-base-uncased")
+
+# Save the model
+model.save_pretrained("./bert2bert")
+
+# Load the model
+model = EncoderDecoderModel.from_pretrained("./bert2bert")
+```
+
+## How to fine-tune the model
```
Comment (stevhliu) [inline docs/source/en/model_doc/encoder-decoder.md:199]:
```suggestion
## Finetune a model
```

File: docs/source/en/model_doc/encoder-decoder.md [hunk 4]
```diff
@@ -178,3 +178,51 @@ were contributed by [ydshieh](https://github.com/ydshieh).
...
+    return_tensors="pt",
+).input_ids
+
+# The forward function automatically creates the correct decoder_input_ids
+loss = model(input_ids=input_ids, labels=labels).loss
+```
+
+## Warning about configuration values
+
+It is important to correctly set the configuration values for the `EncoderDecoderModel`. Incorrect configuration values can lead to unexpected behavior. For more information on how to set the configuration values, refer to the [related issue](https://github.com/huggingface/transformers/issues/15479).
```
Comment (stevhliu) [inline docs/source/en/model_doc/encoder-decoder.md:228]: You can wrap this with the doc-builder's special callout syntax for warnings :)

```suggestion
> [!WARNING]
> It is important to correctly set the configuration values for the [`EncoderDecoderModel`]. Incorrect configuration values can lead to unexpected behavior. For more information on how to set the configuration values, refer to this [related issue](https://github.com/huggingface/transformers/issues/15479).
```

**Review by stevhliu — COMMENTED** [review]
Very nice, thanks for improving the docs! 🤗 

If you can also rebase your branch on main, that should fix the failing `build_pr_documentation` test.

File: docs/source/en/model_doc/encoder-decoder.md [hunk 5]
```diff
@@ -178,3 +178,52 @@ were contributed by [ydshieh](https://github.com/ydshieh).
...
+labels = tokenizer(
+    "the eiffel tower surpassed the washington monument to become the tallest structure in the world. it was the first structure to reach a height of 300 metres in paris in 1930. it is now taller than the chrysler building by 5. 2 metres ( 17 ft ) and is the second tallest free - standing structure in paris.",
+    return_tensors="pt",
+).input_ids
+
+# The forward function automatically creates the correct decoder_input_ids
+loss = model(input_ids=input_ids, labels=labels).loss
+```
+
+## Warning about configuration values
```
Comment (stevhliu) [inline docs/source/en/model_doc/encoder-decoder.md:226]: This can be removed

```suggestion
```
